### ISSUE TITLE
Catalog price rules are not included in CartItemPrices

### ISSUE THREAD

**Issue by ghost** [magento/magento2#26738]
### Preconditions (*)

1. v2.3.4

### Steps to reproduce (*)

1. Create a catalog price rule (e.g. 20% off of gear)
2. Add an applicable item to cart
3. Do a graphql query
```graphql
query {
  cart(cart_id: "...") {
    items {
        prices {
          discounts {
            amount {
              value
            }
            label
          }
          price {
            value
          }
          row_total {
            value
          }
          row_total_including_tax {
            value
          }
          total_item_discount {
            value
          }
        }
    }
  }
}
```

### Expected result (*)

1. Receive price values that include catalog price discount

### Actual result (*)

1. Receive price values that don't include discount

### Additional info from engcom
We have created a product with a price of 100 and applied a discount of 20%
From the Storefront, the added products price is **80** + 5 shipping, correct :+1: 

The cart query displays the wrong :x:  ```price: 100``` ```discount: null```
```graphql
{
  "data": {
    "cart": {
      "items": [
        {
          "prices": {
            "discounts": null,   #<-----
            "price": {
              "value": 100   #<-----
            },
            "row_total": {
              "value": 100
            },
            "row_total_including_tax": {
              "value": 100
            },
            "total_item_discount": {
              "value": 0
            }
          }
        }
      ]
    }
  }
}
```

**Comment by shikhamis11** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11, here is your [Magento instance](https://i-26738-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26738-2-4-develop.instances.magento-community.engineering/admin_dd0e](https://i-26738-2-4-develop.instances.magento-community.engineering/admin_dd0e)
Login: `00734d83` Password: `64d8dddad96c`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-32120` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by prabhuram93** [comment]
Hi @AndrewBarnhart! Our team has started working on this in the scope of MC-32120.

**Comment by magento-engcom-team** [comment]
Hi @AndrewBarnhart, @lenaorobei, @shikhamis11, @prabhuram93. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.4-develop` branch(es).
Related commit(s):
 - [80a321de0c054ef546478e2b741d6eb1ddb4a78e](https://github.com/magento/magento2/commit/80a321de0c054ef546478e2b741d6eb1ddb4a78e)
 - [8b63f456e6179f45ac394406eca38244f27da6ae](https://github.com/magento/magento2/commit/8b63f456e6179f45ac394406eca38244f27da6ae)
 - [dbfd32c826c425bba51fbc12cd20c66249833764](https://github.com/magento/magento2/commit/dbfd32c826c425bba51fbc12cd20c66249833764)
 - [10501a6b270fb6a0a7217269209c9d3fa00bbd0d](https://github.com/magento/magento2/commit/10501a6b270fb6a0a7217269209c9d3fa00bbd0d)
 - [9ce87f83c60723f082c2616c004e886dd6c2e76e](https://github.com/magento/magento2/commit/9ce87f83c60723f082c2616c004e886dd6c2e76e)
 - [9692fbc5c98ce31af9eb859b971d7f1dbb472c8b](https://github.com/magento/magento2/commit/9692fbc5c98ce31af9eb859b971d7f1dbb472c8b)
 - [162b5095f7d3f1291c6522694706d7ff2ca3ab8f](https://github.com/magento/magento2/commit/162b5095f7d3f1291c6522694706d7ff2ca3ab8f)

The fix will be available with the upcoming `2.4.0` release.

**Comment by magento-engcom-team** [comment]
Hi @AndrewBarnhart, @lenaorobei, @shikhamis11, @prabhuram93. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-32120` was closed as `Fixed`. The fix was delivered into `magento/magento2:2.4-develop` branch(es).
Related commit(s):
 - [80a321de0c054ef546478e2b741d6eb1ddb4a78e](https://github.com/magento/magento2/commit/80a321de0c054ef546478e2b741d6eb1ddb4a78e)
 - [8b63f456e6179f45ac394406eca38244f27da6ae](https://github.com/magento/magento2/commit/8b63f456e6179f45ac394406eca38244f27da6ae)
 - [dbfd32c826c425bba51fbc12cd20c66249833764](https://github.com/magento/magento2/commit/dbfd32c826c425bba51fbc12cd20c66249833764)
 - [10501a6b270fb6a0a7217269209c9d3fa00bbd0d](https://github.com/magento/magento2/commit/10501a6b270fb6a0a7217269209c9d3fa00bbd0d)
 - [9ce87f83c60723f082c2616c004e886dd6c2e76e](https://github.com/magento/magento2/commit/9ce87f83c60723f082c2616c004e886dd6c2e76e)
 - [9692fbc5c98ce31af9eb859b971d7f1dbb472c8b](https://github.com/magento/magento2/commit/9692fbc5c98ce31af9eb859b971d7f1dbb472c8b)
 - [162b5095f7d3f1291c6522694706d7ff2ca3ab8f](https://github.com/magento/magento2/commit/162b5095f7d3f1291c6522694706d7ff2ca3ab8f)

The fix will be available with the upcoming `2.4.2` release.
